pkg/discovery/worker/metrics_collector.go

```diff
@@ -189,6 +190,55 @@ func collectContainersComputeResources(pod *v1.Pod) (float64, float64, float64,
 	return totalCPULimits, totalCPURequests, totalMemLimits, totalMemRequests
 }
 
+func (collector *MetricsCollector) CollectPodVolumeMetrics() []*repository.PodVolumeMetrics {
+	var podVolumeMetricsCollection []*repository.PodVolumeMetrics
+
+	var podToVolsMap map[string][]repository.MountedVolume
+	if collector.Cluster.PodToVolumesMap != nil {
+		podToVolsMap = collector.Cluster.PodToVolumesMap
+	}
+
+	metricsSink := collector.MetricsSink
+	//Iterate over all the pods in the collection
+	for _, pod := range collector.PodList {
+		podKey := fmt.Sprintf("%s/%s", pod.Namespace, pod.Name)
```

Comment:
There's a func `util.PodKeyFunc(pod)` which is equivalent to the logic here.

Reply:
Updated!